realm/realm-library/src/syncIntegrationTest/kotlin/io/realm/SyncedRealmIntegrationTests.kt

```diff
@@ -0,0 +1,308 @@
+        val randomizer = Random()
+        for (i in 0..9) {
+            val task = Realm.getInstanceAsync(config, object : Realm.Callback() {
+                override fun onSuccess(realm: Realm) { fail() }
+                override fun onError(exception: Throwable) { fail(exception.toString()) }
+            })
+            SystemClock.sleep(randomizer.nextInt(5).toLong())
+            task.cancel()
+        }
+        // Leave some time for the async callbacks to actually get through
+        looperThread.postRunnableDelayed(
+                Runnable { looperThread.testComplete() },
+                1000
+        )
+    }
+
+    @Test
+    fun waitForInitialRemoteData_readOnlyTrue() {
+        // 1. Copy a valid Realm to the server (and pray it does it within 10 seconds)
+        val configOld: SyncConfiguration = configurationFactory.createSyncConfigurationBuilder(user, user.id)
+                .testSchema(SyncStringOnly::class.java)
+                .build()
+        Realm.getInstance(configOld).use { realm ->
+            realm.executeTransaction { realm ->
+                for (i in 0..9) {
+                    realm.createObject(SyncStringOnly::class.java, ObjectId()).chars = "Foo$i"
+                }
+            }
+            realm.syncSession.uploadAllLocalChanges()
+        }
+        user.logOut()
+
+        // 2. Local state should now be completely reset. Open the Realm again with a new configuration which should
+        // download the uploaded changes (pray it managed to do so within the time frame).
+        // Use different user to trigger different path
+        val user2 = app.registerUserAndLogin(TestHelper.getRandomEmail(), SECRET_PASSWORD)
+        val configNew: SyncConfiguration = configurationFactory.createSyncConfigurationBuilder(user2, user.id)
+                .waitForInitialRemoteData()
+                .readOnly()
+                .testSchema(SyncStringOnly::class.java)
+                .build()
+        assertFalse(configNew.testRealmExists())
+        Realm.getInstance(configNew).use { realm ->
+            assertEquals(10, realm.where(SyncStringOnly::class.java).count())
+        }
+        user.logOut()
+    }
+
+    @Test
+    // FIXME
+    @Ignore("Not really sure how to do this test with new sync")
+    fun waitForInitialRemoteData_readOnlyTrue_throwsIfWrongServerSchema() {
+        val configNew: SyncConfiguration = configurationFactory.createSyncConfigurationBuilder(user, user.id)
+                .waitForInitialRemoteData()
+                .readOnly()
+                .testSchema(SyncStringOnly::class.java)
+                .build()
+        assertFalse(configNew.testRealmExists())
+        assertFailsWith<RealmMigrationNeededException> {
+            // This will fail, because the server Realm is completely empty and the Client is not allowed to write the
+            // schema.
+            // FIXME Does not throw. How to test schema migration with new sync when server is in dev mode
```

Comment:
We cannot test this in dev mode, since the server will just create the schema.
Maybe if we have a model class with a specific property that has a different type on the the client than on the server.

We can add this class as part of the server setup?

Reply:
Seems to work if the class is just now known on the server, so introduce a class for this purpose.